### PR TITLE
Add bank tests for accounts_data_size and resize transactions / rent collection

### PR DESCRIPTION
**Pull request by brooksprumo**
#### Problem

There are not explicit accounts data size tests to ensure that:
1. Resize/realloc transactions correctly update the bank's accounts data size fields
2. Reclaimed accounts from rent collection correctly update the bank's accounts data size fields

#### Summary of Changes

Add two tests. One for accounts data size and resize transactions, and one for accounts data size and rent collection.

### CODE REVIEW

**Comment by brooksprumo** [comment]
@jackcmay Tagging you for review here since I used your `test_resize_and_rent()` as the basis of `test_accounts_data_size_and_resize_transactions()` here.

**Comment by brooksprumo** [comment]
@jeffwashington If you have cycles, can you take a look at `test_accounts_data_size_and_rent_collection()` here? Should be straight forward.

**Review by jeffwashington — APPROVED** [review]
yay more tests!

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -18891,4 +18892,131 @@ pub(crate) mod tests {
...
+                account_balance,
+                mock_program_id,
+                recent_blockhash,
+            );
+            let result = bank.process_transaction(&transaction);
+            assert!(result.is_ok());
+            let accounts_data_size_after = bank.load_accounts_data_size();
+            assert_eq!(
+                accounts_data_size_after,
+                accounts_data_size_before + account_grow_size as u64,
```
Comment (Lichtso) [inline runtime/src/bank.rs:18946]: You are using `saturating_sub()` so `saturating_add()` here for consistency?

**Review by Lichtso — DISMISSED** [review]

File: runtime/src/bank.rs (hunk 1 above)
Comment (brooksprumo) [reply runtime/src/bank.rs]: Done!

**Pull request merged**